oak_idl/tests/oak_abi.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use oak_idl::*;
+
+// An example of a (subset of) the Oak ABI expressed as an Oak Service.
+//
+// Note that this service is exposed on a single channel handle, and the numbers correspond to
+// method identifiers within that channel.
+service! {
+    Oak {
+        42 => fn lookup_data(key: Vec<u8>) -> Option<Vec<u8>>;
```

Comment:
For this mapping from (channel) handle to extensions/service we need to make sure it is in sync with the runtime too.

I believe the handles would roughly take the place of the `*_ABI_FUNCTION_NAME` (e.g., https://github.com/project-oak/oak/blob/a978dc57fec21d62bcbe0d4f59e0d97245ef2e96/oak_functions/loader/src/lookup.rs#L31) in the OakFunctions runtime.

We'll also need to make sure it is in sync with the (feature-gated) configuration of binary of the runtime, i.e., if we expose `report_metric` as a service the binary needs to be compiled with feature=oak-metrics.